featurebyte/service/feature_table_cache.py

```diff
@@ -443,11 +504,105 @@ async def create_or_update_feature_table_cache(
             )
+
+    async def create_view_from_cache(
+        self,
+        feature_store: FeatureStoreModel,
+        observation_table: ObservationTableModel,
+        graph: QueryGraph,
+        nodes: List[Node],
+        output_view_details: TableDetails,
+        is_target: bool = False,
+        feature_list_id: Optional[PydanticObjectId] = None,
+        serving_names_mapping: Optional[Dict[str, str]] = None,
+        progress_callback: Optional[
+            Callable[[int, Optional[str]], Coroutine[Any, Any, None]]
+        ] = None,
+    ) -> None:
+        """
+        Create create or update cache table and create a new view which refers to the cached table
+
+        Parameters
+        ----------
+        feature_store: FeatureStoreModel
+            Feature Store object
+        observation_table: ObservationTableModel
+            Observation table object
+        graph: QueryGraph
+            Graph definition
+        nodes: List[Node]
+            Input node names
+        output_view_details: TableDetails
+            Output table details
+        is_target : bool
+            Whether it is a target computation call
+        feature_list_id: Optional[PydanticObjectId]
+            Optional feature list id
+        serving_names_mapping: Optional[Dict[str, str]]
+            Optional serving names mapping if the observations set has different serving name columns
+            than those defined in Entities
+        progress_callback: Optional[Callable[[int, Optional[str]], Coroutine[Any, Any, None]]]
+            Optional progress callback function
+        """
+        await self.create_or_update_feature_table_cache(
+            feature_store=feature_store,
+            observation_table=observation_table,
+            graph=graph,
+            nodes=nodes,
+            is_target=is_target,
+            feature_list_id=feature_list_id,
+            serving_names_mapping=serving_names_mapping,
+            progress_callback=progress_callback,
+        )
+        cache_metadata = (
+            await self.feature_table_cache_metadata_service.get_or_create_feature_table_cache(
+                observation_table_id=observation_table.id,
+            )
+        )
+        cached_features = {
+            feature.definition_hash: feature.feature_name
+            for feature in cache_metadata.feature_definitions
+        }
+        hashes = await self.nodes_to_defition_hashes(graph, nodes)
+
+        db_session = await self.session_manager_service.get_feature_store_session(
+            feature_store=feature_store
+        )
+
+        columns_expr = [
+            expressions.alias_(
+                quoted_identifier(cast(str, cached_features[definition_hash])),
+                alias=graph.get_node_output_column_name(node.name),
+                quoted=True,
+            )
+            for definition_hash, node in hashes.items()
+        ]
+        select_expr = expressions.select(*columns_expr).from_(
```

Comment:
We should include the row index column `InternalName.TABLE_ROW_INDEX` in the view.